### DISCUSSION TITLE
Don't work GPIO by Switch on ESP32 DevKit V1

### DISCUSSION THREAD

**Discussion by kurogedelic** [RavenSystem/esp-homekit-devices#2113] [MEPLHAA Scripts]
**Detailed steps to reproduce it**
This is my first time trying this so I'm very confused.
I wrote a simple script using DevKit V1 with DHT22 and Digital Out.
The script was successfully uploaded via WiFi and the device was successfully added to "My Home".
It appears that data is being sent from DHT22 for "Temperature and Humidity".
However, even if I toggle the switch in the "Home" app, the specified GPIO 12 does not work.
I thought the Script and board numbers were wrong and tried all the GPIOs but none were outputting.
What's weird is that if I put my ear close to the DevKit V1 and toggle the switch, I can hear a jittery noise coming from the chip. In other words, HAA itself probably succeeded in changing the state of ESP32's GPIO.
I have two boards, so this is not a board failure.
Maybe I'm making a basic misunderstanding?

**MEPLHAA Script**
`{
  "c": {
    "o": 0,
        "l": 2, 
            "i": 1  
  },
  "a": [
    {
      "t": 24,
      "g": 13,
      "n": 2
    },
    {
      "t": 1,
      "0": {
        "r": [
     [12,0]
        ]
      },
         "1": {
        "r": [
     [12,1]
        ]
      }
    }
  ]
}`

**Used device**
ESP32 DevKit V1

**HAA version**
12.7.3

**iOS/iPadOS version**
iOS16.5/17 RC

**Logs**
is it this?
`Reboot
22:17:04.217 -> ets Jun  8 2016 00:22:57
22:17:04.217 -> 
22:17:04.217 -> rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
22:17:04.217 -> configsip: 0, SPIWP:0xee
22:17:04.217 -> clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
22:17:04.249 -> mode:DIO, clock div:2
22:17:04.249 -> load:0x3fff0030,len:152
22:17:04.249 -> load:0x40078000,len:12868
22:17:04.249 -> load:0x40080400,len:4
22:17:04.249 -> load:0x40080404,len:2908
22:17:04.249 -> entry 0x4008056c`

**Comment by RavenSystem** [comment]
"io" array is missed, and needed for GPIO 12: https://github.com/RavenSystem/esp-homekit-devices/wiki/gpios-configuration

**Comment by Rahulsharma0810** [comment]
try below code, DHT22 is connected on GPIO4

```
{
  "c": {
    "n": "Some Name",
    "sn": "SN",
    "tz": "IST-5:30",
    "o": 1,
    "r": [
      { "n":0, "s":115200, "p":0, "b":0, "g":[1] }
    ],
    "io": [[4]],
    "f": 1
  },
  "a": [
          {"t": 24,"n": 2,"g": 4}
        ]
}
```

**Comment by kurogedelic** [comment]
I added the gpio array and moved the DHT22 GPIO witch using @Rahulsharma0810  's settings as a reference.
It's working fine for now.
Thanks!

```
{
	"c": {
		"io": [[[12, 13], 2], [4]],
		"l": 13,
		"b": [[0, 5]],
		"f": 1
	},
	"a": [
		{
			"t": 24,
			"n": 2,
			"g": 4,
			"k": 10
		},
		{
			"0": {
				"r": [[12, 0]]
			},
			"1": {
				"r": [[12, 1]]
			},
			"t": 1,
			"d": 3600,
			"s": 5,
			"b": [[0]]
		}
	]
}

```
